src/librustdoc/test.rs

```diff
@@ -314,49 +241,73 @@ fn run_test(
     };
     let output_file = outdir.path().join("rust_out");
 
-    let config = interface::Config {
-        opts: sessopts,
-        crate_cfg: config::parse_cfgspecs(cfgs),
-        input,
-        input_path: None,
-        output_file: Some(output_file.clone()),
-        output_dir: None,
-        file_loader: None,
-        diagnostic_output: DiagnosticOutput::Raw(box Sink(data.clone())),
-        stderr: Some(data.clone()),
-        crate_name: None,
-        lint_caps: Default::default(),
-    };
+    let mut compiler = Command::new(std::env::current_exe().unwrap().with_file_name("rustc"));
```

Comment:
I'm pretty sure this will fail on OpenBSD, see #61377.

Reply:
any suggestions how to solve this? I do not have access to an OpenBSD computer

Reply:
IMO, the correct thing here is probably to fallback to `RUSTC` from the environment, and then falling back to just `rustc` (whichever one we find in PATH). This isn't great, but I think should work fine for most cases.

Reply:
tried to fix it like this but I get a lot of tests that fails due to RUSTC is set stage0 while the rustdoc that is tested is stage2, but it feels more correct to check the RUSTC env var first otherwise the miri case will still not work 

    let mut compiler = if let Some(rustc) = env::var_os("RUSTC"){
        eprintln!("env var {:?}",&rustc);
        Command::new(&rustc)
    }else {
        let rustc = std::env::current_exe().unwrap_or_default().with_file_name("rustc");
        eprintln!("exec {:?}",&rustc);
        Command::new(&rustc)
    };